CMakeLists.txt

```diff
@@ -1,69 +1,91 @@
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # Set default values for variables/options
-set(CONFIG "assert" CACHE STRING "Default configuration")
-set(UMD_HOME "$ENV{TT_METAL_HOME}/tt_metal/third_party/umd")
+set(UMD_HOME "${CMAKE_SOURCE_DIR}/tt_metal/third_party/umd")
 
-option(ENABLE_PROFILER "Enable profiler" OFF)
-option(ENABLE_TRACY "Enable Tracy" OFF)
 option(ENABLE_CODE_TIMERS "Enable code timers" OFF)
 option(TT_METAL_VERSIM_DISABLED "Disable TT_METAL_VERSIM" ON)
-option(TT_METAL_CREATE_STATIC_LIB "Create static library" OFF)
 
-set(CMAKE_CXX_STANDARD 17)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
+# Default to building everything as a shared lib
+if($ENV{TT_METAL_CREATE_STATIC_LIB})
+    option(BUILD_SHARED_LIBS "Create shared library" OFF)
+else()
+    option(BUILD_SHARED_LIBS "Create shared library" ON)
+endif()
+message(STATUS "Build shared libs: ${BUILD_SHARED_LIBS}")
 
 ############################################
 # Constructing interface libs for common compiler flags, header directories, and libraries
 #   These interface libs are linked with PUBLIC scope at lowest common target (tt_metal/common) and at tt_metal_libs level
-#   in order to propogate to the rest of tt_metal, tt_eager and cpp tests.
+#   in order to propogate to the rest of tt_metal, tt_eager, etc.
 ############################################
 add_library(metal_common_libs INTERFACE)
-target_link_libraries(metal_common_libs INTERFACE dl z boost_thread boost_filesystem boost_system boost_regex pthread atomic hwloc stdc++)
+target_link_libraries(metal_common_libs INTERFACE
+    dl z pthread atomic stdc++  # system libraries
+    Boost::thread Boost::filesystem Boost::system Boost::regex hwloc    # hwloc has no cmake support, find_package won't find it
+)
 
-# note on flags:
+# Note on flags:
 #   DFMT_HEADER_ONLY must be for every target or else they won't interact with the header only fmt as intended
+#   ttnn and tt_lib will break if built with LTO, so leaving -fno-lto in compile options
 add_library(compiler_warnings INTERFACE)
 target_compile_options(compiler_warnings INTERFACE -Werror -Wdelete-non-virtual-dtor -Wreturn-type -Wswitch -Wuninitialized -Wno-unused-parameter)
 
 add_library(compiler_flags INTERFACE)
 target_link_libraries(compiler_flags INTERFACE compiler_warnings)
-target_compile_options(compiler_flags INTERFACE --std=c++17 -MMD -mavx2 -fPIC -DFMT_HEADER_ONLY -fvisibility-inlines-hidden)
+target_compile_options(compiler_flags INTERFACE --std=c++17 -MMD -mavx2 -fPIC -DFMT_HEADER_ONLY -fvisibility-inlines-hidden -fno-lto)
 
 if(TT_METAL_VERSIM_DISABLED)
     target_compile_options(compiler_flags INTERFACE -DTT_METAL_VERSIM_DISABLED)
 endif()
 if(ENABLE_CODE_TIMERS)
     target_compile_options(compiler_flags INTERFACE -DTT_ENABLE_CODE_TIMERS)
 endif()
-if(ENABLE_PROFILER)
+if($ENV{ENABLE_PROFILER})
+    message(STATUS "PROFILER ENABLED")
     target_compile_options(compiler_flags INTERFACE -DPROFILER)
 endif()
-if(ENABLE_TRACY)
+if($ENV{ENABLE_TRACY})
```

Comment:
why is this an env variable now?

Reply:
Made it an env so could be set through the pip install as well: `ENABLE_TRACY=1 pip install `.

I could change it to how `TT_METAL_CREATE_STATIC_LIB` is used, so devs could set an env or directly change the underlyinng cmake option with `-DTRACY_ON` or smthn

Reply:
I'm not sure how much we should support using tracy through wheel - it seems to be something that people should be using through source?

If we do want to support this, I think we should verify compile options like this in this call and pass them in as build option strings into subprocess.run rather than a blind passthrough of env vars into CMake. Other examples of this in the open source world include Apple's MLX `setup.py` - may have mentioned in another comment.